### PR TITLE
[Experimental] Add QUIC support and use netpoll for more efficient network handling

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <kpango@vdaas.org>

### Description:
Vald has been working on minimizing the overhead of the network by enhancing the libraries around each network, including gRPC.
Recently, the final specification of QUIC has been fixed, and Vald would like to verify the approach of gRPC over QUIC, since it will be in practical use soon.
This PR contains the code to verify gRPC over QUIC using quic-go from netpoll and other Network I/O blocking solutions.

### Environment:

- Vald Version: v1.7.13
- Go Version: v1.23.2
- Rust Version: v1.81.0
- Docker Version: v27.3.1
- Kubernetes Version: v1.31.1
- Helm Version: v3.16.2
- NGT Version: v2.2.4
- Faiss Version: v1.9.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

## Release Notes

- **New Features**
  - Introduced support for QUIC protocol connections, enhancing network capabilities across various components.
  - Added a `network` property in multiple configurations, allowing users to specify network types (TCP, UDP, UNIX) for better flexibility.
  - Implemented a new method to retrieve detailed index information, enhancing functionality in the operation package.
  - Added `unhealthyPodEvictionPolicy` settings in various configurations to manage pod eviction based on health status.

- **Bug Fixes**
  - Improved error handling and logging in various components, ensuring clarity during connection attempts and shutdown processes.

- **Documentation**
  - Updated configuration files and schemas to reflect new network settings and options.

- **Tests**
  - Enhanced test coverage and robustness, particularly in handling Kubernetes resources and validating connection behaviors.

### CODE REVIEW

**Comment by kpango** [comment]
@coderabbitai fullreview

**Review by kmrmt — APPROVED** [review]

**Review by vankichi — APPROVED** [review]
LGTM

**Review by vankichi — DISMISSED** [review]
Sorry,  I found CI errors, please wait for merging to the main.

File: internal/client/v1/client/discoverer/discover.go [hunk 1]
```diff
@@ -343,21 +354,39 @@ func (c *client) discoverNodes(ctx context.Context) (nodes *payload.Info_Nodes,
...
+			if l > maxPodLen {
+				maxPodLen = l
+			}
+			slices.SortFunc(nodes.Nodes[i].Pods.Pods, func(left, right *payload.Info_Pod) int {
+				return cmp.Compare(left.GetMemory().GetUsage(), right.GetMemory().GetUsage())
+			})
 		}
 	}
+	nbody, err := nodes.MarshalJSON()
+	if err != nil {
```
Comment (kmrmt) [inline internal/client/v1/client/discoverer/discover.go:387]:
```suggestion
	if err == nil {
```

**Review by kmrmt — CHANGES_REQUESTED** [review]

File: internal/client/v1/client/discoverer/discover.go (hunk 1 above)
Comment (kpango) [reply internal/client/v1/client/discoverer/discover.go:387]:
```suggestion
	if err == nil && nbody != nil {
```

File: internal/net/quic/listener.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	stream, err := sess.AcceptStream(l.ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &Conn{
+		Connection: sess,
+		Stream:     stream,
+	}, nil
+}
```
[the remark this answers is not in the document]
Comment (kpango) [reply internal/net/quic/listener.go:75]: should we think about io.EOF?

**Review by kmrmt — DISMISSED** [review]

**Review by kmrmt — DISMISSED** [review]

**Review by vankichi — APPROVED** [review]

**Review by kmrmt — APPROVED** [review]

**Review by kmrmt — APPROVED** [review]

**Review by vankichi — APPROVED** [review]

**Pull request merged**